### ISSUE TITLE
browser.visit() breaks

### ISSUE THREAD

**Issue by beshur** [assaf/zombie#751]
Hi.

I'm using Zombie with Mocha to test my project.

In one of the tests I'm making a `browser.visit()` call. In fact it works and does everything that it should, but also throws such an error which eventually breaks the test (also breaks subsequent timeout):

```
    Uncaught TypeError: Cannot read property '2' of null
     at EventQueue.clearTimeout (/home/linups/Documents/ezlo/euser-on-express/tests/node_modules/zombie/lib/zombie/eventloop.js:277:24)
     at SocketConnection.processReply (<anonymous>:147:11)
     at WebSocket.SocketConnection.socket.onmessage (<anonymous>:31:9)
     at WebSocket.onMessage (/home/linups/Documents/ezlo/euser-on-express/tests/node_modules/zombie/node_modules/ws/lib/WebSocket.js:360:18)
     at WebSocket.EventEmitter.emit (events.js:98:17)
     at Receiver.self._receiver.ontext (/home/linups/Documents/ezlo/euser-on-express/tests/node_modules/zombie/node_modules/ws/lib/WebSocket.js:697:1
)
     at Receiver.opcodes.1.finish (/home/linups/Documents/ezlo/euser-on-express/tests/node_modules/zombie/node_modules/ws/lib/Receiver.js:397:14)
     at Receiver.expectHandler (/home/linups/Documents/ezlo/euser-on-express/tests/node_modules/zombie/node_modules/ws/lib/Receiver.js:384:31)
     at Receiver.add (/home/linups/Documents/ezlo/euser-on-express/tests/node_modules/zombie/node_modules/ws/lib/Receiver.js:93:24)
     at Socket.firstHandler (/home/linups/Documents/ezlo/euser-on-express/tests/node_modules/zombie/node_modules/ws/lib/WebSocket.js:678:22)
     at Socket.EventEmitter.emit (events.js:95:17)
     at Socket.<anonymous> (_stream_readable.js:746:14)
     at Socket.EventEmitter.emit (events.js:92:17)
     at emitReadable_ (_stream_readable.js:408:10)
     at emitReadable (_stream_readable.js:404:5)
     at readableAddChunk (_stream_readable.js:165:9)
     at Socket.Readable.push (_stream_readable.js:127:10)
     at TCP.onread (net.js:526:21)
```

Using `browser.clickLink()` instead now.

**Comment by xbill82** [comment]
Hi @assaf I don't understand the sentence "the timer wouldn't be there *is* you call `browser.destroy`"... The "is" is confusing me: do you mean "the timer wouldn't be there *if* you call...", which would imply that `browser.destroy` must be called after the `visit` call?
By the way, which timer are you referring to? We are experiencing a similar problem using zombie in conjunction with socket.io.

**Comment by assaf** [comment]
TL;DR don't expect the browser to do anything after you destroy it.

The browser will not wait for anything to happen on the socket, the sockets are not yet integrated into the event loop.

**Comment by xbill82** [comment]
Oh, that's too bad.
I understand now why we are experiencing this issue. We'll have to move our tests to phantomjs or some other testing framework (hoping it will support websockets).
In the meanwhile I would be glad to help out for everything that concerns websocket support. I really have no idea about what needs to be done, but if you are ok, I can open an issue where you could drop some clues.

Again, thanks for the wonderful work on zombie.
